test-e2e-re/lib/FixtureUtils.re

```diff
@@ -0,0 +1,84 @@
+module Path = EsyLib.Path;
+module Option = EsyLib.Option;
+module Result = EsyLib.Result;
+module System = EsyLib.System;
+module Dir = Bos.OS.Dir;
+module File = Bos.OS.File;
+module BPath = Bos.OS.Path;
+
+open Result.Syntax;
+
+type fixtureCopy = {
+  name: string,
+  path: string,
+};
+
+type fixtureFile = {
+  name: string,
+  data: string,
+};
+
+type fixtureDir('t) = {
+  name: string,
+  items: list('t),
+};
+
+type fixture =
+  | FixtureSymlink(fixtureCopy)
+  | FixtureFileCopy(fixtureCopy)
+  | FixtureFile(fixtureFile)
+  | FixtureDir(fixtureDir(fixture));
+
+let transfer = (ic, oc, ()) => {
+  let chunkSize = 1024 * 1024; /* 1mb */
+  let buffer = Bytes.create(chunkSize);
+  let rec loop = () => {
+    switch (input(ic, buffer, 0, chunkSize)) {
+    | 0 => Result.return()
+    | n =>
+      output(oc, buffer, 0, n);
+      loop();
+    };
+  };
+  loop();
+};
+
+let copyFile = (iFile, oFile) => {
+  let%bind l1 =
+    File.with_ic(
+      iFile,
+      (ic, ()) => File.with_oc(oFile, transfer(ic), ()),
+      (),
+    );
+  /**
+    Returning a result here, with unwrap outer layer
+    with_oc -> ((result, error), error)
+    with_ic -> (((result, error), error), error)
+    We are 3 layers deep
+   */
+  let%bind result = l1;
+  result;
+};
+
+let rec layoutSync = (initialize, p) =>
+  fun
+  | FixtureSymlink(spec) =>
+    BPath.symlink(Path.v(spec.path), Path.addSeg(p, spec.name))
+  | FixtureFileCopy(spec) =>
+    copyFile(Path.v(spec.path), Path.addSeg(p, spec.name))
+  | FixtureFile(spec) => File.write(Path.addSeg(p, spec.name), spec.data)
+  | FixtureDir(spec) => {
+      let newPath = Path.addSeg(p, spec.name);
+      let%bind _ = Dir.create(newPath);
+      initialize(newPath, spec.items);
+    };
+
+let rec initialize = p =>
+  fun
+  | [] => Result.return()
+  | [item, ...rest] => {
+      switch (layoutSync(initialize, p, item)) {
+      | Ok(_) => initialize(p, rest)
+      | Error(e) => Result.error(e)
```

Comment:
`Result.Syntax` module which opened above contains `error` so it can be just `error()`:
```suggestion
      | Error(e) => error(e)
```